### PR TITLE
Add multiple compute environments and use them!

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

#2699 

## Purpose/Implementation Notes

Ended up being a good sized change, but I'm happy with how this is setup.

There's one env var set in one place (variables.tf) that controls how many job queues there are, then everything else just iterates over them. There's also job queues for smasher jobs and compendia jobs.

I've changed the message queue to decide which job queue a job goes into, so that logic doesn't get dispersed across the project again.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

With the current settings, the end to end tests use multiple compute environments. It seems to be working pretty well and I'm tweaking it as I continue to test  it.

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by kurtwheeler** [comment]
I think I finally tweaked the queue depth tracking correctly, this is the kinda distribution I was hoping for:
image

File: common/data_refinery_common/message_queue.py [hunk 1]
```diff
@@ -33,6 +35,196 @@
...
+JOB_QUEUE_DEPTHS = {}
+for queue_name in settings.AWS_BATCH_QUEUE_ALL_NAMES:
+    JOB_QUEUE_DEPTHS[queue_name] = 0
+
+DOWNLOADER_JOB_QUEUE_DEPTHS = {}
+for queue_name in settings.AWS_BATCH_QUEUE_WORKERS_NAMES:
+    DOWNLOADER_JOB_QUEUE_DEPTHS[queue_name] = 0
+
+
+def count_jobs_in_queue(batch_job_queue) -> int:
```
Comment (davidsmejia) [inline common/data_refinery_common/message_queue.py:53]: would it make more sense to have the name reflect the return value(s)

File: scripts/kill_all_jobs.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ cd "$script_directory" || exit
 while read -r row; do
     # Exporting an expansion rather than a variable, which is exactly what we want to do.
```
Comment (davidsmejia) [inline scripts/kill_all_jobs.sh:11]: is this comment still relevant?

**Review by davidsmejia — APPROVED** [review]
just had some small comments but LGTM

File: common/data_refinery_common/message_queue.py (hunk 1 above)
Comment (wvauclain) [reply common/data_refinery_common/message_queue.py:53]: The type declaration is also wrong, because this function returns a pair of ints

**Pull request merged**